infra/lambda/lambda_ses.py

```
@@ -0,0 +1,283 @@
+
+
+def format_scan_timestamp(timestamp: str, timezone_name: str) -> str:
+    """Parse an ISO timestamp and return a human-readable string in the given timezone.
+
+    Naive timestamps (no tz info) are treated as UTC.
+    Raises ValueError for invalid timestamps or unknown timezone names.
+    """
+    try:
+        tz = ZoneInfo(timezone_name)
+    except (ZoneInfoNotFoundError, KeyError):
+        raise ValueError(f"Unknown timezone: {timezone_name!r}")
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)